src/panels/config/devices/ha-config-device-page.ts

```diff
@@ -441,35 +453,37 @@ export class HaConfigDevicePage extends LitElement {
             ${this._related?.automation?.length
               ? html`
                   <div class="items">
-                    ${this._related.automation.map((automation) => {
-                      const entityState = this.hass.states[automation];
-                      return entityState
-                        ? html`<ha-tooltip
-                            placement="left"
-                            .disabled=${!!entityState.attributes.id}
-                            .content=${this.hass.localize(
-                              "ui.panel.config.devices.cant_edit"
-                            )}
-                          >
-                            <a
-                              href=${ifDefined(
-                                entityState.attributes.id
-                                  ? `/config/automation/edit/${encodeURIComponent(entityState.attributes.id)}`
-                                  : undefined
+                    ${this._toEntities(this._related.automation).map(
```

Comment:
We should wrap this in memoized, so we don't do the sorting and filtering on every render.

Reply:
Refactored to memoize. Hope I did it right..